Adafruit_SHT31.cpp

```diff
@@ -189,8 +189,8 @@ bool Adafruit_SHT31::readTempHum(void) {
   int32_t stemp = (int32_t)(((uint32_t)readbuffer[0] << 8) | readbuffer[1]);
   // simplified (65536 instead of 65535) integer version of:
   // temp = (stemp * 175.0f) / 65535.0f - 45.0f;
-  stemp = ((4375 * stemp) >> 14) - 4500;
-  temp = (float)stemp / 100.0f;
+  stemp = ((175.0f * (float)stemp) / 65535.0f) - 45.0f;
+  temp = (float)stemp;
```

Comment:
what you're doing here is floor(temp), because you first assign a temperature in celcius(!) to stemp, which is an integer variable (so it contains values like 19, 20, 21) and then you convert it to a float in the line here:

``` 
   temp = (float) stemp
```

afterwards you have the clipped temperature value as float, so 19.0, 20.0, 21.0 etc. according to the datasheet, the temperature sensor has a resolution of 0.015 °C (accuracy < 0.2°C) so clipping to full digit temperature values actually *introduces* an error of theoretically up to 1°C.  I don't think thats what you wanted to do.